poet/src/build_project/mod.rs

```diff
@@ -197,13 +191,13 @@ pub async fn build_project(
 
     // Validate before/after/parent documents in collections
     for reference in content_document_by_basename.values() {
-        for author_basename in &reference.front_matter.authors {
-            if !authors.contains_key(author_basename) {
-                error_collection.register_error(
-                    reference.basename().to_string(),
-                    anyhow!("Author does not exist: '{author_basename}'"),
-                );
-            }
+        let (_, not_found) = authors.resolve(&reference.front_matter.authors);
```

Comment:
You are calling `.resolve` twice, and each time you ignore one of the results. You can instead do that just once instead (probably in the `.par_iter` part)